src/test/java/api/handlers/CloseAgedToLostLoanWhenLostItemFeesAreClosedApiTests.java

```diff
@@ -0,0 +1,127 @@
+import static org.folio.circulation.support.JsonPropertyFetcher.getUUIDProperty;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import org.folio.circulation.support.http.client.IndividualResource;
+import org.junit.Before;
+import org.junit.Test;
+
+import api.support.APITests;
+import api.support.builders.ItemBuilder;
+import io.vertx.core.json.JsonObject;
+import lombok.val;
+
+public class CloseAgedToLostLoanWhenLostItemFeesAreClosedApiTests extends APITests {
+  private IndividualResource loan;
+  private IndividualResource item;
+
+  @Before
+  public void createLoanAndAgeToLost() {
+    feeFineOwnerFixture.cd1Owner();
+    feeFineTypeFixture.lostItemFee();
+    feeFineTypeFixture.lostItemProcessingFee();
+
+    val result = ageToLostFixture.createLoanAgeToLostAndChargeFees(
+      lostItemFeePoliciesFixture.ageToLostAfterOneMinutePolicy()
+        .withName("Age to lost policy")
+        .withSetCost(10.0)
+        .chargeItemAgedToLostProcessingFee(15.00));
+
+    item = result.getItem();
+    loan = result.getLoan();
+  }
+
+  @Test
+  public void shouldCloseLoanWhenAllFeesClosed() {
+    feeFineAccountFixture.payLostItemFee(loan.getId());
+    feeFineAccountFixture.payLostItemProcessingFee(loan.getId());
+
+    eventSubscribersFixture.publishLoanRelatedFeeFineClosedEvent(loan.getId());
+
+    assertThat(loansFixture.getLoanById(loan.getId()).getJson(), isClosed());
+    assertThat(itemsClient.getById(item.getId()).getJson(), isLostAndPaid());
+  }
+
+  @Test
+  public void shouldDisregardNonLostFeeTypes() {
+    feeFineAccountFixture.payLostItemFee(loan.getId());
+    feeFineAccountFixture.payLostItemProcessingFee(loan.getId());
+
+    final IndividualResource manualFee = feeFineAccountFixture
+      .createManualFeeForLoan(loan, 10.00);
+
+    eventSubscribersFixture.publishLoanRelatedFeeFineClosedEvent(loan.getId());
+
+    assertThat(loansFixture.getLoanById(loan.getId()).getJson(), isClosed());
+    assertThat(itemsClient.getById(item.getId()).getJson(), isLostAndPaid());
+
+    assertThat(accountsClient.getById(manualFee.getId()).getJson(), isOpen());
+  }
+
+  @Test
+  public void shouldNotCloseLoanWhenProcessingFeeIsNotClosed() {
+    feeFineAccountFixture.payLostItemFee(loan.getId());
+
+    eventSubscribersFixture.publishLoanRelatedFeeFineClosedEvent(loan.getId());
+
+    assertThat(loansFixture.getLoanById(loan.getId()).getJson(), isOpen());
+    assertThat(itemsClient.getById(item.getId()).getJson(), isAgedToLost());
+  }
+
+  @Test
+  public void shouldNotCloseLoanIfSetCostFeeIsNotClosed() {
+    feeFineAccountFixture.payLostItemProcessingFee(loan.getId());
+
+    eventSubscribersFixture.publishLoanRelatedFeeFineClosedEvent(loan.getId());
+
+    assertThat(loansFixture.getLoanById(loan.getId()).getJson(), isOpen());
+    assertThat(itemsClient.getById(item.getId()).getJson(), isAgedToLost());
+  }
+
+  @Test
+  public void shouldNotCloseLoanIfActualCostFeeShouldBeCharged() {
```

Comment:
How are loans closed where there are actual cost fees to be charged?

Reply:
As far as I know, there is no way to close such loans. This decision has been made by Emma when we developed Declared lost feature.